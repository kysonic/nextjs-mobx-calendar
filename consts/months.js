export const DAYS = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
export const MONTHS = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
export const DAYS_IN_MONTH = 31;
export const DAYS_IN_WEEK = 7;
