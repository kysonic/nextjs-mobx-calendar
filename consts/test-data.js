export const APP_INITIAL_DATA = {
    navigationStore: {
        items: [
            {
                "id": "home",
                "title": "Home",
                "href": "/"
            }
        ]
    },
    dateStore: {
        ISODate: '2019-01-01T00:00:00.000Z'
    }
};

export const CALENDAR_INITIAL_DATA = {
    "2019-04-10T00:00:00.000Z": [
        {
            "id": "abcde1",
            "title": "Action 1",
            "description": "Description of action 1"
        },
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        }
    ],
    "2019-03-11T00:00:00.000Z": [
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        },
        {
            "id": "abcde3",
            "title": "Action 4",
            "description": "Description of action 4"
        }
    ]
}
