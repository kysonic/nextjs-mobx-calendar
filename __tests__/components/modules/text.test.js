import Text from '../../../components/modules/text/text';
import renderer from 'react-test-renderer';
import {render, cleanup} from 'react-testing-library';
import TextStore from '../../../stores/text';
import {TEXT_DATA} from '../../../consts/test-data';
import 'jest-dom/extend-expect';
import axios from 'axios';

jest.mock('axios');

axios.get.mockResolvedValueOnce({data: TEXT_DATA});

describe('Test Text component', () => {
    afterEach(cleanup);

    test('Snap Testing', () => {
        const tree = renderer.create(<Text textStore={ {text: "TEST TEXT"} } />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('UI Testing', async () => {
        const textStore = new TextStore({ text: "TEST TEXT"});
        const { getByTestId } = render(<Text textStore={textStore} />);
        expect(getByTestId('text')).toHaveTextContent('TEST TEXT');
        await textStore.fetch();
        expect(axios.get).toHaveBeenCalledTimes(1);
        expect(axios.get).toHaveBeenCalledWith('/static/data/text.json');
        expect(getByTestId('text')).toHaveTextContent('Welcome to the Mobx Calendar');
    });
});
