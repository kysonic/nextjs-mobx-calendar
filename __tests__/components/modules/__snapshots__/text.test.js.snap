// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Text component Snap Testing 1`] = `
<div
  className="jsx-2395214420 text"
  data-testid="text"
>
  TEST TEXT
</div>
`;
